server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1389,12 +1418,39 @@ public static Setting<Integer> intSetting(
         return new Setting<>(
             key,
             Integer.toString(defaultValue),
-            (s) -> parseInt(s, minValue, maxValue, key, isFiltered(properties)),
+            new IntegerParser(key, minValue, maxValue, isFiltered(properties)),
             validator,
             properties
         );
     }
 
+    private static class IntegerParser implements Function<String, Integer>, Writeable {
```

Comment:
You'll need to access this class (and the others in this PR) from `WriteableSetting` when you are reading/writing it across transport, so its visibility must be at least package private.  I don't see a reason not to make them all `public`.   You should also add javadocs explaining why they exist.